data/src/main/java/com/github/leuludyha/data/repository/datasourceImpl/UserDataSourceImpl.kt

```diff
@@ -0,0 +1,27 @@
+package com.github.leuludyha.data.repository.datasourceImpl
+
+import com.github.leuludyha.data.repository.datasource.UserDataSource
+import com.github.leuludyha.domain.model.library.Mocks
+import com.github.leuludyha.domain.model.user.User
+import com.google.firebase.ktx.Firebase
+
+class UserDataSourceImpl(
+    private val firebase: Firebase
+) : UserDataSource {
+
+    override fun getNeighbouringUsersOf(
+        user: User, distance: (User, User) -> Float, n: Int
+    ): List<User> {
+        /*
+            TODO Find the least costly way to compute distance between a significant
+            portion of users in the DB with firebase (Since it is non-relational)
+        */
+//        firebase.firestore.collection("user")
+//            .orderBy()
+//            .limit(n.toLong())
+//            .get()
+        return listOf(Mocks.user)
```

Comment:
Because Firebase isn't a relational database, so we cannot do subqueries so I haven't found a way to apply a custom function to do the sorting in the database. If we do it client-side, it needs to load all users in the database pairwise which is not realistic without pulling a clever trick which I don't have an idea about yet